cmd/commands.go

```diff
@@ -26,49 +25,59 @@ type Command struct {
-	if !cmd.Global {
-		regionFlag := &cli.StringFlag{Name: "region", Value: "", Usage: "Name of the region to use"}
-		cmd.Command.Flags = append(cmd.Command.Flags, regionFlag)
-		action := cmd.Command.Action
-		cmd.Command.Action = func(c *cli.Context) error {
-			token := os.Getenv("SCALINGO_API_TOKEN")
-
-			currentUser, err := config.C.CurrentUser()
-			if err != nil || currentUser == nil {
-				err := session.Login(c.Context, session.LoginOpts{APIToken: token})
-				if err != nil {
-					errorQuit(err)
-				}
-			}
+	// Global commands are simply added to the list of commands
+	if cmd.Global {
+		cmds.commands = append(cmds.commands, cmd.Command)
+		return
+	}
+
+	// Regional commands are modified before being added to the list of commands.
+	regionFlag := &cli.StringFlag{Name: "region", Value: "", Usage: "Name of the region to use"}
+	cmd.Command.Flags = append(cmd.Command.Flags, regionFlag)
+	// I don't really understand the official documentation about this field.
+	// But setting it to true enables the display of the help usage message when `cli.ShowCommandHelp` is called. Without this setting, the call to `cli.ShowCommandHelp` displays a "No help topic for command" error message.
+	cmd.Command.HideHelp = true
+	action := cmd.Command.Action
+	cmd.Command.Action = regionalCommandAction(action)
+}
+
+func regionalCommandAction(action cli.ActionFunc) cli.ActionFunc {
+	return func(c *cli.Context) error {
+		token := os.Getenv("SCALINGO_API_TOKEN")
 
-			regions, err := config.EnsureRegionsCache(c.Context, config.C, config.GetRegionOpts{
-				Token: token,
-			})
+		currentUser, err := config.C.CurrentUser()
+		if err != nil || currentUser == nil {
+			err := session.Login(c.Context, session.LoginOpts{APIToken: token})
 			if err != nil {
 				errorQuit(err)
 			}
-			currentRegion := regionNameFromFlags(c)
+		}
 
-			// Detecting Region from git remote
-			if currentRegion == "" {
-				currentRegion = detect.GetRegionFromGitRemote(c, &regions)
-			}
+		regions, err := config.EnsureRegionsCache(c.Context, config.C, config.GetRegionOpts{
+			Token: token,
+		})
+		if err != nil {
+			errorQuit(err)
+		}
+		currentRegion := regionNameFromFlags(c)
 
-			if config.C.ScalingoRegion == "" && currentRegion == "" {
-				region := getDefaultRegion(regions)
-				debug.Printf("[Regions] Use the default region '%s'\n", region.Name)
-				currentRegion = region.Name
-			}
+		// Detecting Region from git remote
+		if currentRegion == "" {
+			currentRegion = detect.GetRegionFromGitRemote(c, &regions)
+		}
 
-			if currentRegion != "" {
-				config.C.ScalingoRegion = currentRegion
-			}
-			action(c)
+		if config.C.ScalingoRegion == "" && currentRegion == "" {
+			region := getDefaultRegion(regions)
+			debug.Printf("[Regions] Use the default region '%s'\n", region.Name)
+			currentRegion = region.Name
+		}
 
-			return nil
+		if currentRegion != "" {
+			config.C.ScalingoRegion = currentRegion
 		}
+
+		return action(c)
```

Comment:
This is a noticeable change. 

With this change, if an action returns an error, it displays the error message (and that's it). There is no other messages than the error string representation.

Before this PR, if an action returned an error it would simply be ignored.